### PR TITLE
[PT FE]: enable dtype in softmax and constant filling, extend logical…

### PR DESCRIPTION
**Pull request by eaidova**
… ops support

### Details:
 - *extend aten::softmax with dtype support*
 - *support aten::__or__*
 - *fix input signature for aten::empty*
 - *support aten::baddbmm (add matmul for batched matrix)*

P.S. this list of ops required to support bloom model conversion, tested locally that after these changes it successfully converted
### Tickets:
 - *TBD*

### CODE REVIEW

File: src/frontends/pytorch/src/op/bitwise_not.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/logical_not.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_bitwise_not(NodeContext& context) {
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/bitwise_not.cpp:14]:
```suggestion
OutputVector translate_bitwise_not(const NodeContext& context) {
```

Comment (rkazants) [reply src/frontends/pytorch/src/op/bitwise_not.cpp:14]: I remember Maxim should have added this. Please check

File: src/frontends/pytorch/src/op/bitwise_not.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    num_inputs_check(context, 1, 2);
+    auto x = context.get_input(0);
+    FRONT_END_OP_CONVERSION_CHECK(x.get_element_type().is_dynamic() || x.get_element_type() == element::boolean,
+                                  "aten::bitwise_not suppored only for boolean input");
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/bitwise_not.cpp:18]:
```suggestion
    FRONT_END_OP_CONVERSION_CHECK(x.get_element_type().compatible(element::boolean),
                                  "aten::bitwise_not suppored only for boolean input");
```

File: src/frontends/pytorch/src/utils.cpp [hunk 3]
```diff
@@ -142,6 +142,22 @@ element::Type convert_dtype(int64_t pt_type) {
     return TORCH_TO_OV_TYPE.at(pt_type);
 };
 
+Output<Node> apply_dtype(const NodeContext& context, int dtype_port, Output<Node> input_tensor) {
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.cpp:145]:
```suggestion
Output<Node> apply_dtype(const NodeContext& context, int dtype_port, const Output<Node>& input_tensor) {
```

File: src/frontends/pytorch/src/utils.hpp [hunk 4]
```diff
@@ -41,6 +41,9 @@ std::shared_ptr<Node> get_axes_range(const NodeContext& context, int input_id);
 std::shared_ptr<Node> numel(const NodeContext& context, const Output<Node>& x);
 
 element::Type convert_dtype(int64_t dtype_value);
+
+Output<Node> apply_dtype(const NodeContext& context, int dtype_port, Output<Node> input_tensor);
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.hpp:45]:
```suggestion
Output<Node> apply_dtype(const NodeContext& context, int dtype_port, const Output<Node>& input_tensor);
```

File: src/frontends/pytorch/src/op/bitwise_not.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/bitwise_not.cpp:14]: it is still not merged due to segfault issue in tests

Comment (eaidova) [reply src/frontends/pytorch/src/op/bitwise_not.cpp:14]: for now, input mutation does not work if I will use const context

File: src/frontends/pytorch/src/op/bitwise_not.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/bitwise_not.cpp]: updated, thanks

File: src/frontends/pytorch/src/utils.cpp (hunk 3 above)
Comment (eaidova) [reply src/frontends/pytorch/src/utils.cpp]: updated

File: src/frontends/pytorch/src/utils.cpp [hunk 5]
```diff
@@ -142,6 +142,22 @@ element::Type convert_dtype(int64_t pt_type) {
...
+        auto dtype = convert_dtype(context.const_input<int64_t>(dtype_port));
+        return context.mark_node(std::make_shared<opset10::Convert>(input_tensor, dtype));
+
+    } else if (const auto& fw_node = cast_fw_node(context.get_input(dtype_port).get_node_shared_ptr(), "prim::dtype")) {
+        auto out_tensor = fw_node->input_value(0);
+        return context.mark_node(std::make_shared<opset10::ConvertLike>(input_tensor, out_tensor));
+
+    } else {
+        FRONT_END_OP_CONVERSION_CHECK(false, "Couldn't get dtype input");
+    }
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.cpp:157]: why can't we check if `context.get_input(dtype_port).get_element_type().is_static()` and use `Convert` if it is static? And use `ConvertLike` in all other cases?

Can you please see (may be it will be helpful) `ov::Output<ov::Node> create_same_type_const_scalar(const ov::Output<ov::Node>& same_type_output, T value)` approach from TF FE? Can we apply the same?

File: src/frontends/pytorch/src/utils.cpp [hunk 6]
```diff
@@ -142,6 +142,22 @@ element::Type convert_dtype(int64_t pt_type) {
     return TORCH_TO_OV_TYPE.at(pt_type);
 };
 
+Output<Node> apply_dtype(const NodeContext& context, int dtype_port, const Output<Node>& input_tensor) {
+    if (std::dynamic_pointer_cast<opset10::Constant>(
+            context.get_input_from_visible_context(dtype_port).get_node_shared_ptr())) {
+        auto dtype = convert_dtype(context.const_input<int64_t>(dtype_port));
+        return context.mark_node(std::make_shared<opset10::Convert>(input_tensor, dtype));
+
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.cpp:150]:
```suggestion
```

File: src/frontends/pytorch/src/utils.cpp [hunk 7]
```diff
@@ -142,6 +142,22 @@ element::Type convert_dtype(int64_t pt_type) {
...
+    } else if (const auto& fw_node = cast_fw_node(context.get_input(dtype_port).get_node_shared_ptr(), "prim::dtype")) {
+        auto out_tensor = fw_node->input_value(0);
+        return context.mark_node(std::make_shared<opset10::ConvertLike>(input_tensor, out_tensor));
+
```
Comment (rkazants) [inline src/frontends/pytorch/src/utils.cpp:154]:
```suggestion
```

**Review by rkazants — APPROVED** [review]
left some comments.
Other looks good to me.

Also, please fix the build failure due to type check

File: src/frontends/pytorch/src/utils.cpp (hunk 5 above)
Comment (eaidova) [reply src/frontends/pytorch/src/utils.cpp:157]: it works slightly different, dtype can be represented as operation torch.dtype(tensor), which returns dtype of input tensor or can be represented by native torch dtype e.g. torch.float32, which turns to enum constant int and should be mapped to openvino dtype. As we do not have operations which extract dtype from node, we are trying to resolve torch.dtype with replacement convertlike with its input.

However, maybe I do not fully understand your suggestion, maybe it required offline discussion

File: src/frontends/pytorch/src/op/softmax.cpp [hunk 8]
```diff
@@ -5,18 +5,23 @@
 #include "openvino/op/softmax.hpp"
 
 #include "openvino/frontend/pytorch/node_context.hpp"
+#include "pt_framework_node.hpp"
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/op/softmax.cpp]: removed, thanks

**Review by mvafin — APPROVED** [review]
Please resolve conflict
